### PR TITLE
Update rubocop + lint

### PR DESCRIPTION
**Pull request by manuelpuyol**
Updated:
- rubocop (0.82.0)
- rubocop-github (0.16.0)
- rubocop-performance (1.7.1)
- rubocop-rails (2.5.2)

Linted errors and tuned some rules

### CODE REVIEW

File: test/primer/classify_test.rb [hunk 1]
```diff
@@ -341,7 +341,7 @@ def refute_generated_class(input)
     assert_nil(Primer::Classify.call(**input)[:class])
   end
 
-  def assert_allocations(count, within:, &block)
```
Comment (joelhawksley) [inline test/primer/classify_test.rb:344]: We don't need this argument? I would have thought we did. cc @BlakeWilliams

Comment (manuelpuyol) [reply test/primer/classify_test.rb:344]: I think we don't need it since we use `yield`

File: Gemfile [hunk 2]
```diff
@@ -3,7 +3,7 @@
 source "https://rubygems.org"
 
 gemspec
-rails_version = "#{ENV['RAILS_VERSION'] || '6.0.3.3'}"
+rails_version = (ENV["RAILS_VERSION"] || "6.0.3.3").to_s
```
Comment (joelhawksley) [inline Gemfile:6]: Mind updating this to `6.1.0`? We can also do so in another PR ❤️

**Review by joelhawksley — APPROVED** [review]

File: test/primer/classify_test.rb (hunk 1 above)
Comment (BlakeWilliams) [reply test/primer/classify_test.rb:344]: I forgot where I picked it up, but sometimes `&block` is added to method arguments to signal that they accept a block without requiring API consumers to look at the source.

I'm not opposed to removing it though, if that's what our linter recommends.

File: Gemfile (hunk 2 above)
Comment (manuelpuyol) [reply Gemfile:6]: I can do it in a separate PR <3
